sdk/keyvault/keyvault-certificates/tsconfig.json

```diff
@@ -22,6 +22,6 @@
     "resolveJsonModule": true
   },
   "compileOnSave": true,
-  "exclude": ["node_modules", "./samples/**/*.ts"],
-  "include": ["./src/**/*.ts", "./test/**/*.ts"]
+  "exclude": ["node_modules", "../keyvault-common/node_modules", "./samples/**/*.ts"],
+  "include": ["./src/**/*.ts", "./test/**/*.ts", "../keyvault-common/**/*.ts"]
```

Comment:
My TS/JS knowledge is limited but by doing this does it cause npm to pull in the compiled JS code from keyvault-common into the keyvault package? 

Assuming it does that will it correctly scope/isolate the code across the various keyvault packages so they don't conflict? 

Also I assume that none of this common code will have any shared state across the various keyvault packages?

Reply:
Questions! Let me see:

> doing this does it cause npm to pull in the compiled JS code from keyvault-common into the keyvault package?

Yep

> will it correctly scope/isolate the code across the various keyvault packages so they don't conflict?

It will be the same code for the three of them. If one package doesn't like it, it will complain, so this code needs to be good for all of the three of them at the same time.

> Also I assume that none of this common code will have any shared state across the various keyvault packages?

No state will be shared 🌞

Reply:
Thanks for helping me better understand how this shared code model works for JS/TS.

One more clarifying question:

>It will be the same code for the three of them. If one package doesn't like it, it will complain, so this code needs to be good for all of the three of them at the same time.

Am I correct in assuming that there will be an independent copy of this code for each keyvault library an application is using? Assuming that is the case I guess there is a balancing act that we need to maintain about the size of this common library such that we don't want it to become too large as it will bloat an application. I suspect if it reaches some tipping point we then make it a shared library instead of shared duplicated code.

Reply:
It's exactly as you're assuming, this code will be copied into each package's artifact (the compressed output that NPM generates).

It's indeed a balancing act. I myself was proponent of not doing this and going directly into separate packages, but that adds considerable overhead regarding version management and releases.

We're also not preemptively moving everything into this folder. So far I'm not planning to move the generated code here. Only this file, and potentially some test utilities (that don't end up with the final package).